snark-verifier/src/loader/halo2/shim.rs

```diff
@@ -419,10 +420,7 @@ mod halo2_wrong {
                     Ok::<_, Error>((scalar.deref().clone(), self.assign_constant(ctx, *base)?))
                 })
                 .collect::<Result<Vec<_>, _>>()?;
-            let pairs = pairs
```

Comment:
clippy fix: complain removing `map` but we still need ref-of-tuple to tuple-of-ref